packages/ingest/test/test-ftp.js

```diff
@@ -12,59 +12,62 @@ const {
   headObject
 } = require('@cumulus/common/aws');
 const { randomString } = require('@cumulus/common/test-utils');
-const { ftpMixin: TestFtpMixin } = require('../ftp');
-
-class MyTestDiscoveryClass {
-  constructor(useList) {
-    this.decrypted = true;
-    this.host = '127.0.0.1';
-    this.password = 'testpass';
-    this.path = '';
-    this.provider = { encrypted: false };
-    this.useList = useList;
-    this.username = 'testuser';
-  }
-}
+const FtpProviderClient = require('../FtpProviderClient');
 
 test('useList is present and true when assigned', async (t) => {
   const jsftpSpy = sinon.spy(JSFtp);
-  const { ftpMixin } = proxyquire('../ftp', {
+  const ProxiedFtpProviderClient = proxyquire('../FtpProviderClient', {
     jsftp: jsftpSpy
   });
 
-  class MyTestFtpDiscoveryClass extends ftpMixin(MyTestDiscoveryClass) {}
-  const myTestFtpDiscoveryClass = new MyTestFtpDiscoveryClass(true);
+  const myFtpProviderClient = new ProxiedFtpProviderClient({
+    host: '127.0.0.1',
+    username: 'testuser',
+    password: 'testpass',
+    path: '',
+    useList: true
+  });
 
-  await myTestFtpDiscoveryClass.list();
+  await myFtpProviderClient.list();
 
   t.is(jsftpSpy.callCount, 1);
   t.is(jsftpSpy.getCall(0).args[0].useList, true);
 });
 
 test('useList defaults to false when not assigned', async (t) => {
   const jsftpSpy = sinon.spy(JSFtp);
-  const { ftpMixin } = proxyquire('../ftp', {
+  const ProxiedFtpProviderClient = proxyquire('../FtpProviderClient', {
```

Comment:
Probably I don't understand how `proxyquire` works, but why do we have to recreate this proxied client in each test?

Reply:
We have to create it each time because we're using a different `jsftpSpy` in each test, then counting the number of times that spy was invoked.

Reply:
I'm pretty sure you could use the same spy/proxied client across all of your tests and do `spy.resetHistory()` in the `beforeEach` to reset the call count for each test